lms/services/organization.py

```diff
@@ -47,6 +47,18 @@ def get_by_public_id(self, public_id: str) -> Optional[List]:
             self._db_session, Organization, public_id, region=self._region
         )
 
+    def search(self, name):
+        """
+        Search organizations.
+
+        :param name: Match organization by name. Case insensitive.
+        """
+        return (
+            self._db_session.query(Organization)
+            .filter(Organization.name.ilike(f"%{name}%"))
```

Comment:
This is not a very sophisticated text search but I reckon it would go a long way.